src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/AWSSecretsManagerPlugin.java

```diff
@@ -130,47 +142,104 @@ public AWSSecretsManagerPlugin(
 
   @Override
   public void openInitialConnection(ConnectionUrl connectionUrl) throws SQLException {
-    final Pair<String, Region> secretKey = Pair.of(secretId, region);
+
     final Properties properties = new Properties();
     properties.putAll(connectionUrl.getOriginalProperties());
-    Secret secret = SECRET_CACHE.get(secretKey);
+
+    boolean secretWasFetched = updateSecret(false);
```

Comment:
I don't think we need to return a boolean here. Just return the secret

Reply:
secretWasFetched helps to identify whether another fetch of credentials is needed when login attempt failed.